react/features/base/jwt/functions.ts

```diff
@@ -75,26 +78,24 @@ function isValidUnixTimestamp(timestamp: number | string) {
 }
 
 /**
- * Returns a list with all validation errors for the given jwt.
+ * Returns a map with all validation errors for the given jwt.
  *
  * @param {string} jwt - The jwt.
- * @returns {Array<string>} - An array containing all jwt validation errors.
+ * @returns {Object} - An object containing all jwt validation errors.
  */
 export function validateJwt(jwt: string) {
-    const errors: string[] = [];
-
-    if (!jwt) {
-        return errors;
-    }
-
+    let errors: Object = {};
     const currentTimestamp = new Date().getTime();
 
     try {
         const header = jwtDecode(jwt, { header: true });
         const payload = jwtDecode(jwt);
 
         if (!header || !payload) {
-            errors.push('- Missing header or payload');
+            errors = {
+                ...errors,
+                '- Missing header or payload': i18n.t('dialog.tokenAuthFailedReason.headerOrPayload')
```

Comment:
Not a fan of this. We could instead return an array or constants, and then render the translated messages in the notification, can't we?

I imagine something like:

```
const JWT_VALIDATION_ERRORS: {
    MISSING_HEADER: 'missing-header',
    MISSING_PAYLOAD: 'missing-payload',
    MISSING_KID: 'missing-kid',
    ...
};
```

Generally speaking we try not to pass translated strings around, but to translate them right where they are rendered.

Reply:
> to translate them right where they are rendered.
Yes, that was my initial approach, with the enum like object. But then I needed arguments to my description arguments and I started twisting the Notification.tsx implementation to translate arguments and stumbled into the scenario in which we might translate something we don't want translated (your occasional odd username). I figured this is the safest approach.

I chose to keep the keys from the initial implementation to display them directly in the logs, because I found it odd to translate that part. WDYT?

Reply:
Can't the translation happen at the time of dispatch then?

Reply:
You mean in the middleware?

Reply:
And have the localized keys as the value in the `JWT_VALIDATION_ERRORS` object?

Reply:
Yes in the middleware, where we dispatch the showNotification action.

> And have the localized keys as the value in the `JWT_VALIDATION_ERRORS` object?

I don't think you can do this because the i18n backend needs to be initialized and it won't be at import time so you can't define a constant.

Reply:
I meant to have what you described above with the keys instead:
```
const JWT_VALIDATION_ERRORS: {
    MISSING_HEADER: 'headerNotFound',
    MISSING_PAYLOAD: 'headerNotFound',
    MISSING_KID: 'kidNotFound',
    ...
};
```

And compose the locale key in the dispatch as you said, by iterating over the errors and calling something like t(\`dialog.tokenAuthFailedReason.${error[idx]}\`)

Reply:
Yeah, makes sense!